### PR TITLE
Add VueJS layer

### LINKED ISSUE #7: Add a VueJS layer & mock some functions
This is going to be a fairly simple first part of the issue: 

- [ ] Add a VueJS layer

What I mean here is that I'll add a reference to the public VueJS library ( as seen in the tutorials here: https://laracasts.com/series/learn-vue-2-step-by-step )

For the second part of this issue I want to mock some custom functions in a JavaScript file. By this I mean to create the function names themselves, without making them work. Knowing which functions we will need ahead of time is part of good application planning and also lends itself to Test Driven Development (we'll want to have some tests in place as the complexity of the app grows - more on that in the future).

Here are some of the functions we'll need:
- [ ] validate spaces (should probably decide on a naming convention, `spaces` or `squares`?)
- [ ] move chip
- [ ] capture chip
- [ ] double-capture
- [ ] reset board

### PR DESCRIPTION
**Pull request by ghost**
fixes #7 

I added the VueJS reference to the checkers-flex.html file and used a snippet of hard-coded data in the same file to demo and mess with VueJS variables in the page.

During that process I realized we will want the chips object to be an array, rather than an object.

I updated the `chips_state.json` file to reflect that adjustment to the 2 instances of a chips object: `player1.chips` and `player2.chips`. Within each item in the arrays (now they're each an array of objects), I added the key `chipId` to be used in the markup for identifying each chip in each array.

See my comments in the *files changed* diff

### CODE REVIEW

File: checkers-flex.html [hunk 1]
```diff
@@ -6,11 +6,14 @@
   <link rel="stylesheet" href="checkers-flex-style.css" type="text/css">
 </head>
 <body>
+  <div id="game">
+  <h4>Player 1: {{ player1.name }}, chip color: {{ player1.color }}</h4>
+  <h4>Player 2: {{ player2.name }}, chip color: {{ player2.color }}</h4>
```
Comment (ghost) [inline checkers-flex.html:11]: here I am adding some player stats, and making sure that my Vue instance is reading the data by outputting the variables in *mustache syntax*: `{{ player1.name }}`

File: checkers-flex.html [hunk 2]
```diff
@@ -6,11 +6,14 @@
...
 
-  <div class ="wrapper">
+  <div class="wrapper">
     <!--8th row -->
     <div class="row">
-      <div class="box red-square"><div class="piece black"></div></div>
+      <div class="box red-square"></div>
```
Comment (ghost) [inline checkers-flex.html]: I removed the black chip from row 8 and used it to make duplicates in row 1

File: checkers-flex.html [hunk 3]
```diff
@@ -87,15 +90,95 @@
...
+        <div class="piece black" :id="player1.chips[2].chipId"></div>
+      </div>
+
       <div class="box red-square"></div>
-      <div class="box black-square" id="g1" nextValidSquare="['f2', 'h2']"></div>
+
+      <div class="box black-square" id="g1" nextValidSquare="['f2', 'h2']">
+        <div class="piece black" :id="player1.chips[3].chipId"></div>
+      </div>
+
```
Comment (ghost) [inline checkers-flex.html]: here I've added the first 4 chips from player1. I am using the colon at the beginning of the `id` attribute to indicate that I want to bind my Vue data to the attribute. The colon there is shorthand for v-binding, and lets the compiler know that the content in the quotes, `player1.chips[3].chipId` is a Vue data variable, much like the *mustache syntax* above. VueJS lets us do some fun things with HTML tag attributes, but HTML attributes don't like mustache syntax, so there is this v-binding shorthand to work with.

If you checkout this branch and load this file in your browser, then inspect the HTML here, you will see that the individual black chips have ids like `black-1`, `black-2`, etc.

File: checkers-flex.html [hunk 4]
```diff
@@ -87,15 +90,95 @@
...
   </div>
+</div>
+
+  <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
+  <script>
+
+    new Vue({
+      el: '#game',
+      // data: chips :: when the chips_state.json can be imported, we can store it in a variable, such as `chips`, but for this part I will hardcode the data...
+      data: {
```
Comment (ghost) [inline checkers-flex.html]: eventually when we break this all up into `.vue` templates with imported data, you will see how this all can be broken down and then compiled automatically, but for now as we figure out the nuances of our data needs, we can play with the hardcoded object here and backfill changes to the `chips_state.json` file.

**Comment by ghost** [comment]
I've updated the entire board with the default chip states because I think we really need to have it fully setup to start applying the select, validate and movement functions.

**Review by sxkosone — APPROVED** [review]
Thank you for the updates and explanations.

**Pull request merged**